Source/JavaScriptCore/runtime/ImportMap.cpp

```diff
@@ -225,10 +224,47 @@ Expected<void, String> ImportMap::registerImportMap(const SourceCode& sourceCode
         return codePointCompareLessThan(rhs.m_scope.string(), lhs.m_scope.string());
     });
 
+    Integrity integrity;
+    StringBuilder errorMessage;
+    if (auto integrityValue = rootMap->getValue("integrity"_s)) {
+        auto integrityMap = integrityValue->asObject();
+        if (!integrityMap)
+            return makeUnexpected("integrity is not a map"_s);
+
+        // https://html.spec.whatwg.org/C#normalizing-a-module-integrity-map
+        for (auto& [key, value] : *integrityMap) {
+            URL integrityURL = resolve(key, baseURL);
+            if (UNLIKELY(integrityURL.isNull())) {
+                errorMessage.append("Integrity URL "_s);
+                errorMessage.append(key);
+                errorMessage.append(" is not a valid URL starting with '/', './' or '../'\n"_s);
```

Comment:
Isn't this error message a bit misleading given `resolve` also seems to accept a full URL?

Reply:
Added a mention of absolute URLs, thanks! :)